### PR TITLE
Revamp config docs with helm-docs

### PR DESCRIPTION
**Pull request by michaellzc**
This PR added autogen docs based on `values.yaml`.

`charts/<chart_name>/README.md` should no longer be edited, and changes should be made in `charts/<chart_name>/README.md.gotmpl`

View the rendered version here

- https://github.com/sourcegraph/deploy-sourcegraph-helm/tree/02-25-Revamp_config_docs_with_helm-docs/charts/sourcegraph
- https://github.com/sourcegraph/deploy-sourcegraph-helm/tree/02-25-Revamp_config_docs_with_helm-docs/charts/sourcegraph-migrator

## Test plan

Run `./scripts/helm-docs.sh`, you should be able to generate the README.md

close https://github.com/sourcegraph/sourcegraph/issues/31016

### CODE REVIEW

File: charts/sourcegraph-migrator/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph-migrator/values.yaml:27]:
```suggestion
  # learn more from the [Kubernetes documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
```

File: charts/sourcegraph-migrator/README.md.gotmpl
Comment (caugustus-sourcegraph) [inline charts/sourcegraph-migrator/README.md.gotmpl:68]: Should these just be `<serviceName>.affinity` etc. instead of having the leading `.`?

File: charts/sourcegraph-migrator/README.md.gotmpl
Comment (caugustus-sourcegraph) [inline charts/sourcegraph-migrator/README.md.gotmpl:66]: This sentence is a bit confusing since there are no other services in this chart.

File: charts/sourcegraph-migrator/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph-migrator/values.yaml:111]:
```suggestion
  # -- Security context for the `migrator` container,
```

File: charts/sourcegraph/README.md.gotmpl
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/README.md.gotmpl:77]: Same as above - should this leading `.` be removed?

File: charts/sourcegraph/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/values.yaml:160]: Maybe we should remove this setting as an option for the databases and hardcode the count into the template?

File: charts/sourcegraph/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/values.yaml:277]: Uses `view` ClusterRole if set to false, there's no customization currently allowed

File: charts/sourcegraph/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/values.yaml:354]:
```suggestion
  # -- Security context for the `gitserver` container,
```

File: charts/sourcegraph/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/values.yaml:436]:
```suggestion
  # -- Number of `indexed-search` pod
```

File: charts/sourcegraph/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/values.yaml:702]:
```suggestion
    # -- Docker image name for the `redis-exporter` image
    name: "redis_exporter"
```

File: charts/sourcegraph/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/values.yaml:23]:
```suggestion
  # -- Global Affinity,
```
All these `sourcegraph.` settings are global and will apply to all services.

File: charts/sourcegraph/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/values.yaml:814]: The formatting in the chart ends up weird here and in the other storage class descriptions - i.e. ```Enable creation of storageClass Disable if you have your own existing storage class```

Do we need to add a period on the top line or is there a better solution?

File: charts/sourcegraph/values.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/values.yaml:889]: TODO

File: RELEASE.md
Comment (michaellzc) [reply RELEASE.md]: This is tracked here https://github.com/sourcegraph/sourcegraph/issues/32016

File: charts/sourcegraph/values.yaml
Comment (michaellzc) [reply charts/sourcegraph/values.yaml:160]: > Maybe we should remove this setting as an option for the databases and hardcode the count into the template?

I would love that

File: charts/sourcegraph/values.yaml
Comment (michaellzc) [reply charts/sourcegraph/values.yaml:814]: adding a `period` works!

File: charts/sourcegraph/values.yaml
Comment (michaellzc) [reply charts/sourcegraph/values.yaml]: Updated to 

```md
Enable creation of Role and RoleBinding (RBAC). Uses [view](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles) ClusterRole if set to false
```

**Review by caugustus-sourcegraph — APPROVED** [review]
Nice work! Thanks for slogging through all this.

Side note - when you squash commits during an open review, it makes it hard for reviewers to tell what has changed since the PR was first opened. It's a faster and more reliable review cycle if fixes are kept as separate commits until the PR is merged.

**Comment by michaellzc** [comment]
> Nice work! Thanks for slogging through all this.
> 
> Side note - when you squash commits during an open review, it makes it hard for reviewers to tell what has changed since the PR was first opened. It's a faster and more reliable review cycle if fixes are kept as separate commits until the PR is merged.

Good call out! I can imagine it's pretty hard to inspect the diff again for large PR like this.

**Comment by michaellzc** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).


Job Reference: yNqXdfNl9dvIL5h0XluD

**Pull request merged**
